4_setup_crossval/src/cross_validation_functions.R

```diff
@@ -0,0 +1,41 @@
+
+get_nested_gages<-function(gagesii,nav_distance_km){
+  gage_IDs <- as.character(gagesii$ID)
+  COMIDs <- as.character(gagesii$COMID)
+  drainage_areas<-data.frame(gage_IDs=gage_IDs,
+                             drainArea=readNWISsite(siteNumbers=gage_IDs)$drain_area_va)
+  
+   #column names are the downstream gages, rows with non-zero values are upstream gages
+  upstream_df <- data.frame(matrix(ncol = length(gage_IDs), nrow = length(gage_IDs),data=0))
+  colnames(upstream_df) <- rownames(upstream_df) <- gage_IDs
+
+  for (i in 1:length(gage_IDs)){
+    downstream_da<-drainage_areas$drainArea[i]
+    flowline <- navigate_nldi(list(featureSource = "comid", 
+                                       featureID = COMIDs[i]), 
+                                  mode = "upstreamTributaries",
+                                  distance_km=nav_distance_km)
+    ##list of all upstream COMIDs
+    upstream_list <- flowline$UT_flowlines$nhdplus_comid
+    
+    ##gages located on any upstream COMID (includes the COMID of the downstream gage)
+    upstream_gages<-gage_IDs[which(COMIDs %in% upstream_list)]
+    
+    ##if there are more upstream gages than there are unique COMIDs, then 
+    ##there are more than one gage on a COMID.  If this happens, remove any
+    ##gage from the upstream gages list that has a larger drainage area than the
+    ##downstream_da.
```

Comment:
Just to check my understanding, is this happening only when the "downstream" gage is on the same COMID as an "upstream" gage so that "upstream" gage is actually downstream?

Reply:
Yes.  I'm pretty sure this was the cause of the NAs as well. I'm running it again now to check this. Basically, I had been linking everything by COMID, thinking that was unique to each gage but in cases where two gages on one comid occured, it would include that larger gage as an upstream gage of the smaller gage.

Reply:
haha, no worries, it was easy to go through. I'm hoping this will resolve the ratios that were a little greater than 1, but it may not have been the cause for the ratio that was ~5

Reply:
yes it was - that was the first one I looked at.  Basically there were two gages on that reach - one was ~4.5 sq miles and the other was 27 sq miles.